.github/workflows/gradle.yml

```diff
@@ -0,0 +1,99 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  build_on_linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [ '8', '11', '14', '15', '15-site', '15-leak']
+    steps:
+    - uses: actions/checkout@v2
+
+    - if: "!startsWith(matrix.java, '15')"
+      name: Set up JRE ${{ matrix.java }}
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: ${{ matrix.java }}
+        java-package: jre
+
+    - if: "!startsWith(matrix.java, '15')"
```

Comment:
Up to you what order, but you might consider migrating to toolchains first to remove this sort of stuff

https://docs.gradle.org/current/userguide/toolchains.html

For reference in case it's helpful

https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/main-build.yml#L44

Reply:
> but you might consider migrating to toolchains first to remove this sort of stuff

Yeah, it might be a time to migrate our customized stuff to toolchain. Let me take a look at it. Thanks!